File: suites/02_numerical_computation/comparisons/fft.py
def butterfly_cost(size: int) -> int:
    total, stage = 0, 1
    while stage < size:
        i = 0
        while i < size:
            left = ((i * 19) + stage) % 65521
            right = ((i * 23) + (stage * 3)) % 65521
            total += (left + right) - (left - right)
            i += stage * 2
        stage *= 2
    return total

if __name__ == "__main__":
    print(butterfly_cost(16384))

File: suites/02_numerical_computation/comparisons/fft.cpp
#include <cstdio>
static long long butterfly_cost(long long size) {
    long long total = 0, stage = 1;
    while (stage < size) {
        for (long long i = 0; i < size; i += stage * 2) {
            long long left = ((i * 19) + stage) % 65521;
            long long right = ((i * 23) + (stage * 3)) % 65521;
            total += (left + right) - (left - right);
        }
        stage *= 2;
    }
    return total;
}
int main() { printf("%lld\n", butterfly_cost(16384)); return 0; }
